fdm-data/src/fertilizers/index.test.ts

```diff
@@ -1,87 +1,95 @@
-            //     database
-            //   )
-            // await migrateFdmServer(fdm)
+describe("getFertilizersCatalogue", () => {
+    it("should return the SRM catalogue when catalogueName is 'srm'", () => {
+        const expectedCatalogue = getCatalogueSrm()
+        const actualCatalogue = getFertilizersCatalogue("srm")
+        expect(actualCatalogue).toEqual(expectedCatalogue)
+    })
 
-            // Workaround
-            fdm = drizzle({
-                connection: {
-                    user: user,
-                    password: password,
-                    host: host,
-                    port: port,
-                    database: database,
-                },
-                logger: false,
-                schema: schema,
-            })
+    it("should throw an error when an invalid catalogueName is provided", () => {
+        expect(() => getFertilizersCatalogue("invalid-catalogue")).toThrowError(
+            "catalogue invalid-catalogue is not recognized",
+        )
+    })
 
-            // Run migration
-            await migrate(fdm, {
-                migrationsFolder: migrationsFolderPath,
-                migrationsSchema: "fdm-migrations",
-            })
-        } catch (error) {
-            console.error("Failed to setup database:", error)
-            throw error
-        }
+    it("should return a non-empty array for 'srm' catalogue", () => {
+        const catalogue = getFertilizersCatalogue("srm")
+        expect(Array.isArray(catalogue)).toBe(true)
+        expect(catalogue.length).toBeGreaterThan(0)
     })
 
-    afterAll(async () => {
-        try {
-            // Clean up test data
-            await fdm.transaction(async (tx: FdmType) => {
-                await tx.delete(schema.fertilizerPicking).execute()
-                await tx.delete(schema.fertilizersCatalogue).execute()
-            })
-        } catch (error) {
-            console.error("Failed to cleanup:", error)
-            throw error
+    it("should check if all items in the srm catalogue have the correct source", () => {
+        const catalogue = getFertilizersCatalogue("srm")
+        for (const item of catalogue) {
+            expect(item.p_source).toBe("srm")
         }
     })
+})
 
-    it("should throw error if dataset is not recognized", async () => {
-        await expect(
-            extendFertilizersCatalogue(fdm, "not-existing-dataset"),
-        ).rejects.toThrowError(
-            "catalogue not-existing-dataset is not recognized",
-        )
+describe("getCatalogueSrm", () => {
+    const originalSrm = require("./catalogues/srm.json")
+    it("should return an array of CatalogueFertilizerItem", () => {
+        const catalogue = getCatalogueSrm()
+        expect(Array.isArray(catalogue)).toBe(true)
+        for (const item of catalogue) {
+            expect(typeof item).toBe("object")
+            expect(item).toHaveProperty("p_source")
+            expect(item).toHaveProperty("p_id_catalogue")
```

Comment:
does this imply that all properties are required?

Reply:
Yes, for the SRM list, which is tested here. For other lists it can be different